### DISCUSSION TITLE
The mac_pro instance type does not support the selected Xcode version. Please consider using mac_mini_m1 or mac_mini_m2 instead.

### DISCUSSION THREAD

**Discussion by DenizKucukozturk** [codemagic-ci-cd/codemagic-docs#2391] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I am trying to build my app with Xcode 15.2 with Flutter 3.13.6 but build instantly fails.

Here is the selected instance:
<img width="1337" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/51988155/bdc211dd-5320-4886-808a-e046e4b2ec5b">

Interesting thing is I have 2 other very similar workflows (same Xcode version, same instance) that each build different branch of the same app and they are built by Mac mini M1 with Xcode 15.2 successfuly.

### Expected results

Successful build. 

### Actual results

<img width="1264" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/51988155/2102ef13-5995-4402-a461-f79deec1207e">

### Build id (optional)

65ddc9d4078e7bd17199dbad

**Comment by dbrezack** [comment]
This is also affecting my team

**Comment by dtrdic** [comment]
Hello there, @DenizKucukozturk @dbrezack make sure you are running the right workflow and the right branch, 
![Screenshot 2024-02-27 at 14 41 12](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/1433a375-c696-4c14-9f31-5b005d498253)

as you are probably running different one in where the Mac pro instance is selected
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/dbe6f41d-70c7-489d-b734-149b67371e25)

**Reply by DenizKucukozturk** [reply]
@dtrdic I checked it multiple times. Branch and workflow selection is definitely correct.

**Reply by DenizKucukozturk** [reply]
@dtrdic Just now I tried with Xcode 15.1 and it built successfully. I don't have any idea what's going on.

**Reply by dtrdic** [reply]
As I mentioned earlier, it's possible that you selected the incorrect workflow where the Mac Pro machine was chosen, or you might not have saved the changes when editing the instance type. .. But you are confirming it's working now, right?

**Reply by DenizKucukozturk** [reply]
@dtrdic It is not event possible for me to incorrectly select Mac Pro. My only options are these two. These are VMs not actual machines.
<img width="556" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/51988155/3fd8aaa9-f2d6-4a82-ba28-b871275203c1">

It still doesn't work with Xcode 15.2. I dropped Xcode version to 15.1 and it build with a `Machine: Mac mini M1`. When I try with Xcode 15.2 it tries to build with Mac Pro and fails with that error.

**Reply by DenizKucukozturk** [reply]
So there is a strange correlation between Xcode version and machine used for building.

**Reply by dtrdic** [reply]
hey @DenizKucukozturk i'm trying to reproduce your issue but I can't.. can you please share the entire URL, not just the build id so I can take a closer look?

**Reply by dtrdic** [reply]
@DenizKucukozturk just to point out --> `macOs Intel VMs` are `Mac Pro` machines, maybe this was the confusion?
<img width="737" alt="Screenshot 2024-02-28 at 12 25 42" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/c5d54b99-f978-4c3f-be28-8cbc1148739a">

**Reply by DenizKucukozturk** [reply]
@dtrdic I didn't know MacOS Intel VMs were Mac Pro machines. Now that is clear. It is still strange though because I never changed instance type to macOS Intel. I only changed Xcode version.

**Reply by DenizKucukozturk** [reply]
@dtrdic After successfully building with Xcode 15.1 now I changed it back to Xcode 15.2 and it worked. i think problem solved by itself.

**Reply by dtrdic** [reply]
I think the most probable cause was that you mistakenly selected a MacOS Intel VM and attempted to run it with Xcode 15.2. However, the maximum Xcode version that can be run on a MacOS Intel VM (Mac Pro machine) is Xcode 14.2. You can find the machine specification [here](https://docs.codemagic.io/specs/versions-macos/#intel-based-machines).
But the important thing is that everything is working, so I'll close this issue. 
Have a nice day, cheers!
